Dockerfile

```diff
@@ -15,6 +17,37 @@ RUN set -ex; \
     ; \
     rm -rf /var/lib/apt/lists/*
 
+# Build Python 2.7.9 from source
+RUN set -ex; \
+    curl -LO https://www.python.org/ftp/python/2.7.9/Python-2.7.9.tgz; \
+    tar -xzf Python-2.7.9.tgz; \
+    cd Python-2.7.9; \
+    ./configure --enable-shared; \
+    make; \
+    make install; \
+    cd ..; \
+    rm -rf /Python-2.7.9; \
+    rm Python-2.7.9.tgz
```

Comment:
Hmm, this added 4min to the jenkins build, which is unfortunate.

Could we use ubuntu vivid, which seems to have 2.7.9  (http://packages.ubuntu.com/vivid/python) ?

That way we could install from a package without having to build everything. I believe there was another PR for python3 support where @funkyfuture found that using ubuntu made things easier, but I forget the details of that now.